components/patina_smbios/src/manager.rs

```diff
@@ -0,0 +1,2398 @@
+    }
+
+    /// Initialize with a boot_services reference
+    ///
+    /// # Safety
+    ///
+    /// Must be called exactly once during system initialization.
+    /// The boot_services reference must have 'static lifetime.
+    unsafe fn initialize(&self, boot_services: &'static StandardBootServices) {
+        unsafe { *self.boot_services.get() = Some(boot_services) };
+    }
+
+    /// Get the stored boot_services reference
+    ///
+    /// # Safety
+    ///
+    /// Returns None if not initialized
+    #[allow(dead_code)] // Reserved for future diagnostic access to raw boot services
+    unsafe fn get(&self) -> Option<&'static StandardBootServices> {
+        unsafe { *self.boot_services.get() }
+    }
+}
+
+static BOOT_SERVICES: GlobalBootServices = GlobalBootServices::new();
+
+/// Global storage for the SMBIOS manager wrapped in TplMutex
+///
+/// Uses TplMutex for TPL-aware synchronization. When locked, TPL is raised to CALLBACK
+/// level, preventing timer interrupt reentrancy. TPL is automatically restored when
+/// the lock guard is dropped.
+///
+/// # Safety
+///
+/// - The TplMutex is wrapped in UnsafeCell for static initialization
+/// - Access is protected by TplMutex.lock() which raises TPL to CALLBACK
+/// - The manager is initialized once during protocol installation
+/// - The pointer remains valid for the lifetime of the system
+struct GlobalSmbiosManager {
+    manager: UnsafeCell<Option<TplMutex<'static, SmbiosManager, StandardBootServices>>>,
+}
+
+unsafe impl Sync for GlobalSmbiosManager {}
+
+impl GlobalSmbiosManager {
+    const fn new() -> Self {
+        Self { manager: UnsafeCell::new(None) }
+    }
+
+    /// Initialize the global manager with TplMutex protection
+    ///
+    /// # Safety
+    ///
+    /// Caller must ensure this is called only once during system initialization
+    unsafe fn initialize(
+        &self,
+        tpl_mutex: TplMutex<'static, SmbiosManager, StandardBootServices>,
+    ) -> Result<(), SmbiosError> {
+        let ptr = self.manager.get();
+        if unsafe { (*ptr).is_some() } {
+            return Err(SmbiosError::InvalidParameter); // Already initialized
+        }
+        unsafe { *ptr = Some(tpl_mutex) };
+        Ok(())
+    }
+
+    /// Get a reference to the TplMutex (returns None if not initialized)
+    ///
+    /// # Safety
+    ///
+    /// Returns a raw reference to the TplMutex. Caller must call .lock()
+    /// to get TPL-protected access to the manager.
+    unsafe fn get(&self) -> Option<&'static TplMutex<'static, SmbiosManager, StandardBootServices>> {
+        unsafe { (*self.manager.get()).as_ref() }
+    }
+
+    /// Clear the manager (for cleanup on error)
+    ///
+    /// # Safety
+    ///
+    /// Caller must ensure this is only called during error cleanup
+    unsafe fn clear(&self) {
+        unsafe { *self.manager.get() = None };
+    }
+}
+
+static SMBIOS_MANAGER: GlobalSmbiosManager = GlobalSmbiosManager::new();
```

Comment:
Why does this need to be global? This should be the private state of the service being produced. If multiple services are accessing this global state, then make all services be implemented off of the same underlying underlying struct. Regardless, one of the intents of services is their is no need for global state.

The only time we have found that global state is necessary is when we are producing a EFI compatible UEFI protocol that does not consume a pointer to self in it's functions. 

If it is necessary, then I just need to know why, and maybe we can find a solution to not have global static state.